File: app/src/test/java/com/qifan/leetcode/problems/jewelsAndStones/JewelsAndStones.java
package com.qifan.leetcode.problems.jewelsAndStones;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by Qifan on 2018/6/6.
 */
public class JewelsAndStones {
    @Test
    public void test1() throws Exception {
        String J = "aA";
        String S = "aAAbbbb";
        assertEquals(3, numJewelsInStones(J, S));
    }

    @Test
    public void test2() throws Exception {
        String J = "z";
        String S = "ZZ";
        assertEquals(0, numJewelsInStones(J, S));
    }

    private int numJewelsInStones(String J, String S) {
        int index = 0;
        for (int i = 0; i < J.length(); i++) {
            for (int j = 0; j < S.length(); j++) {
                if (J.charAt(i) == S.charAt(j)) {
                    index++;
                }
            }
        }
        return index;
    }
}

File: app/src/test/java/com/qifan/leetcode/problems/jewelsAndStones/JewelsAndStones.kt
package com.qifan.leetcode.problems.jewelsAndStones

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Created by Qifan on 2018/6/6.
 */

class Solution {

    @Test
    @Throws(Exception::class)
    fun test1() {
        val J = "aA"
        val S = "aAAbbbb"
        assertEquals(3, numJewelsInStones(J, S))
    }

    @Test
    @Throws(Exception::class)
    fun test2() {
        val J = "z"
        val S = "ZZ"
        assertEquals(0, numJewelsInStones(J, S))
    }


     private fun numJewelsInStones(J: String, S: String): Int {
        var index = 0
        for (j in J) {
            for (s in S) {
                if (j == s) {
                    index++
                }
            }
        }
        return index
    }
}